components/script/dom/htmldetailselement.rs

```diff
@@ -49,6 +49,10 @@ impl HTMLDetailsElement {
             document,
         )
     }
+
+    pub fn toggle(&self) {
+        self.SetOpen(!self.Open());
```

Comment:
Try adding `self.upcast::<Node>().dirty(NodeDamage::OtherNodeDamage)` and see if that fixes the rendering issues.

Reply:
Thanks a lot! It works!

Reply:
Actually, let's move it to the attribute_mutated function so that doing things like `elem.removeAttribute('open')` or `elem.open = true` from JS also cause the rendering to update.

Reply:
Done (my manual test case works, and the toggleEvent.html WTP test is still "PASS")